packages/react-native/Libraries/Text/Text/RCTTextView.mm

```diff
@@ -16,6 +16,14 @@
 
 #import <QuartzCore/QuartzCore.h>
 
+#if __IPHONE_16_0
```

Comment:
This needs to be changed as it will only work for iOS 16. 


Do you know any better way to conditionally conform to delegate?

Reply:
Take a look at the types of checks in this change, and the ones linked in the notes: https://github.com/facebook/react-native/pull/39549

Reply:
the availability header has some good macros to use and patterns. I deleted some examples in an earlier commit since we bumped minimum OS, hence the links to commits :P

Reply:
Found that `#ifdef __IPHONE_16_0` works best as its also true for iOS 17